sdlc_xmalab/sdlc_xmalab.py

```diff
@@ -0,0 +1,262 @@
+'''A Complete Set of User-Friendly Tools for DeepLabCut-XMAlab marker tracking'''
+# Import packages
+import os
+import warnings
+
+import cv2
+import deeplabcut
+import numpy as np
+import pandas as pd
+from pathlib import Path
+import importlib.resources as pkg_resources
+from ruamel.yaml import YAML
+
+from .analyzer import Analyzer
+from .autocorrector import Autocorrector
+from .network import Network
+from .xma_data_processor import XMADataProcessor
+
+
+def create_new_project(working_dir=None, experimenter="NA", mode="2D"):
+    """Create a new xrommtools project"""
+    working_dir = Path(working_dir) if working_dir is not None else Path.cwd()
+    working_dir.mkdir(exist_ok=True)
+    (working_dir / "trainingdata").mkdir(exist_ok=True)
+    (working_dir / "trials").mkdir(exist_ok=True)
+
+    # Create a fake video to pass into the deeplabcut workflow
+    dummy_video_path = working_dir / "dummy.avi"
+    frame = np.zeros((480, 480, 3), dtype=np.uint8)
+    out = cv2.VideoWriter(str(dummy_video_path), cv2.VideoWriter_fourcc(*"DIVX"), 15, (480, 480))
+    out.write(frame)
+    out.release()
+
+    # Create a new project
+    task = working_dir.name
+    path_config_file = deeplabcut.create_new_project(
+        task,
+        experimenter,
+        [str(dummy_video_path)],
+        str(working_dir) + '/',
+        copy_videos=True
+    )
+
+    config_path = Path(__file__).parent / 'default_config.yaml'
+    yaml = YAML()
+    with config_path.open() as file:
+        config_data = yaml.load(file)
+    
+    config_data.update({"task": task,
+                        "experimenter": experimenter,
+                        "working_dir": str(working_dir),
+                        "path_config_file": path_config_file})
+
+    if mode == "per_cam":
+        task_2 = f"{task}_cam2"
+        path_config_file_2 = deeplabcut.create_new_project(
+            task_2,
+            experimenter,
+            [str(dummy_video_path)],
+            str(working_dir) + '/',
+            copy_videos=True,
+        )
+        config_data["path_config_file_2"] = path_config_file_2
+
+    # Save configuration
+    config_file = working_dir / "project_config.yaml"
+    with config_file.open('w') as config:
+        yaml.dump(config_data, config)
+
+    # Cleanup
+    try:
+        (Path(path_config_file).parent / "labeled-data/dummy").rmdir()
```

Comment:
Does .unlink() remove the file on the host as well?

Reply:
it does seem to remove all files and folders specified

i ran with these lines commented out and all the dummy file and video were still there

then i ran with this code present and all dummy video files and the folder got removed